pkg/server/instancecleaner/cleaner.go

```diff
@@ -0,0 +1,91 @@
+package instancecleaner
+
+import (
+	"context"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/gocardless/draupnir/pkg/exec"
+	"github.com/gocardless/draupnir/pkg/models"
+	"github.com/gocardless/draupnir/pkg/store"
+	"github.com/prometheus/common/log"
+
+	google "google.golang.org/api/oauth2/v1"
+)
+
+type InstanceCleaner struct {
+	done          chan bool
+	logger        log.Logger
+	instanceStore store.InstanceStore
+	executor      exec.Executor
+}
+
+func NewInstanceCleaner(logger log.Logger, instanceStore store.InstanceStore, executor exec.Executor) *InstanceCleaner {
+	done := make(chan bool)
+	return &InstanceCleaner{
+		done:          done,
+		logger:        logger.With("component", "cleaner"),
```

Comment:
My experience has told me you want to decorate the `component` field from the outside, if possible. It's not ideal that you have no other way of constructing an instance cleaner without this component field, or you may even override an already present component field if this is the behaviour.

Bit of a fiddly piece of advice and I'm still not 100% on it, but I think we do this well [here](https://github.com/gocardless/stolon-pgbouncer/blob/master/cmd/stolon-pgbouncer/main.go#L350-L553) as you can easily go from "here is my main method" to "the logs from this unit are decorated like this"

Reply:
Yeah I think this is totally legit. Good call - moving it out.